File: src/main/java/com/ultreon/bubbles/event/bus/AddonEventBus.java
package com.ultreon.bubbles.event.bus;

import com.ultreon.bubbles.BubbleBlaster;
import com.ultreon.bubbles.mod.ModContainer;
import com.ultreon.bubbles.mod.ModList;
import com.ultreon.hydro.event.Event;
import com.ultreon.commons.lang.ICancellable;
import com.ultreon.hydro.event.bus.EventBus;

public class AddonEventBus extends EventBus {
    @Override
    public <T extends Event> boolean post(T event) {
        ModList.get().getContainers().stream().map(ModContainer::getModObject).forEach((addonObject) -> {
            BubbleBlaster.getLogger().info("Sending addon event to: " + addonObject.getNamespace());
            addonObject.getEventBus().post(event);
        });

        return event instanceof ICancellable && ((ICancellable) event).isCancelled();
    }

    @Override
    public void register(Class<?> clazz) {
        throw new UnsupportedOperationException();
    }

    @Override
    public void register(Object o) {
        throw new UnsupportedOperationException();
    }

    @Override
    public void unregister(Class<? extends Event> event, Class<?> clazz) {
        throw new UnsupportedOperationException();
    }

    @Override
    public void unregister(Class<? extends Event> event, Object o) {
        throw new UnsupportedOperationException();
    }

    @Override
    public void unregister(Class<?> clazz) {
        throw new UnsupportedOperationException();
    }

    @Override
    public void unregister(Object o) {
        throw new UnsupportedOperationException();
    }
}

File: src/main/kotlin/qtech/bubbles/event/bus/AddonEventBus.kt
package qtech.bubbles.event.bus

import qtech.bubbles.BubbleBlaster
import qtech.bubbles.mods.loader.ModContainer
import qtech.bubbles.common.mods.ModObject
import qtech.bubbles.common.mods.ModInstance
import qtech.bubbles.event.Event
import qtech.bubbles.event.ICancellable
import java.lang.UnsupportedOperationException
import kotlin.reflect.KClass

class AddonEventBus : EventBus() {
    override fun <T : Event> post(event: T): Boolean {
        BubbleBlaster.instance.modManager.containers.stream().map(ModContainer::obj).forEach { modInstanceObject: ModObject<out ModInstance>? ->
            BubbleBlaster.logger.info("Sending addon event to: " + modInstanceObject!!.modId)
            modInstanceObject.eventBus.post(event)
        }
        return event is ICancellable && (event as ICancellable).isCancelled
    }

    override fun <T : Event> addListener(clazz: Class<T>, handler: (event: T) -> Unit) {
        throw UnsupportedOperationException("Not available.")
    }

    override fun <T : Event> addListenerK(clazz: KClass<T>, handler: (event: T) -> Unit) {
        throw UnsupportedOperationException("Not available.")
    }

    override fun register(`class`: Class<*>) {
        throw UnsupportedOperationException("Not available.")
    }

    override fun register(o: Any) {
        throw UnsupportedOperationException("Not available.")
    }

    override fun registerK(`class`: KClass<*>) {
        throw UnsupportedOperationException("Not available.")
    }

    override fun registerK(o: Any) {
        throw UnsupportedOperationException("Not available.")
    }

    override fun unregister(`class`: Class<*>) {
        throw UnsupportedOperationException("Not available.")
    }

    override fun unregister(o: Any) {
        throw UnsupportedOperationException("Not available.")
    }

    override fun unregisterK(`class`: KClass<*>) {
        throw UnsupportedOperationException("Not available.")
    }

    override fun unregisterK(o: Any) {
        throw UnsupportedOperationException("Not available.")
    }

    override fun unregister(event: Class<out Event>, `class`: Class<*>) {
        throw UnsupportedOperationException("Not available.")
    }

    override fun unregister(event: Class<out Event>, o: Any) {
        throw UnsupportedOperationException("Not available.")
    }

    override fun unregisterK(event: KClass<out Event>, `class`: KClass<*>) {
        throw UnsupportedOperationException("Not available.")
    }

    override fun unregisterK(event: KClass<out Event>, o: Any) {
        throw UnsupportedOperationException("Not available.")
    }
}